rooms.js

```diff
@@ -1242,6 +1247,20 @@ class ChatRoom extends BasicRoom {
 		this.chatRoomData = (options.isPersonal ? null : options);
 		Object.assign(this, options);
 		if (this.auth) Object.setPrototypeOf(this.auth, null);
+		/** @type {Room?} */
+		this.parent = null;
+		if (this.parent) {
+			/** @type {Room} */
+			let main = Rooms(this.parent);
```

Comment:
Either way, to put this on a single line:

    const main = /** @type {Room} */ (Rooms(this.parent));